### PR TITLE
Refactor how variables are scoped

### PR DESCRIPTION
**Pull request by PascalSenn**
Summary of the changes (Less than 80 chars)

- Detail 1
- Detail 2

Addresses #94

### CODE REVIEW

File: src/Backend/src/Authoring.Abstractions/Variables/Models/VariableValueScope.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+using HotChocolate.Types.Relay;
+
+namespace Confix.Authoring;
+
+public abstract record VariableValueScope([property: ID<Environment>] Guid? EnvironmentId)
+{
+    public abstract string Identifier { get; }
```
Comment (RohrerF) [inline src/Backend/src/Authoring.Abstractions/Variables/Models/VariableValueScope.cs:7]: Why use Identifier? Can really see where this is "actually" used

Comment (RohrerF) [reply src/Backend/src/Authoring.Abstractions/Variables/Models/VariableValueScope.cs:7]: looks like a weird ToString override

Comment (PascalSenn) [reply src/Backend/src/Authoring.Abstractions/Variables/Models/VariableValueScope.cs:7]: This is mappen in mongodb as a unique index

**Review by PascalSenn — COMMENTED** [review]
LGTM

**Pull request merged**